src/Controller/AbstractSecurityController.php

```diff
@@ -82,6 +87,70 @@ public function forgotPasswordAction(Request $request)
         return $this->redirectToRoute($this->context . '_security_login_form');
     }
 
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function resetPasswordAction(Request $request)
+    {
+        if ($this->getUser()) {
+            return $this->redirectToRoute($this->context . '_dashboard');
+        }
+
+        if (!$request->query->has('token')) {
+            $this->addFlash('error', 'flash.security.invalid_token');
+
+            return $this->redirectToRoute($this->context . '_security_login_form');
+        }
+
+        try {
+            $token = $this->getTokenManager()->get(Token::RESET_PASSWORD, $request->query->get('token'));
+        } catch (TokenNotFoundException $e) {
+            $this->addFlash('error', 'flash.security.token_not_found');
+            return $this->redirectToRoute($this->context . '_security_login_form');
+        } catch (TokenExpiredException $e) {
+            $this->addFlash('error', 'flash.security.token_expired');
+            return $this->redirectToRoute($this->context . '_security_login_form');
+        } catch (TokenConsumedException $e) {
+            $this->addFlash('error', 'flash.security.token_used');
+            return $this->redirectToRoute($this->context . '_security_login_form');
+        }
+
+        if (!$token) {
```

Comment:
You don't add a flash message ?
`$this->addFlash('error', 'flash.security.token_not_found');`